wallet/manager.go

```diff
@@ -265,6 +272,240 @@ func (m *Manager) Reserve(ids []types.Hash256, duration time.Duration) error {
 	return nil
 }
 
+// Release releases the given ids.
+func (m *Manager) Release(ids []types.Hash256) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	for _, id := range ids {
+		delete(m.used, id)
+	}
+}
+
+// SelectSiacoinElements selects siacoin elements from the wallet that sum to
+// at least the given amount. Returns the elements, the element basis, and the
+// change amount. The selected elements are locked for the given duration.
+func (m *Manager) SelectSiacoinElements(walletID ID, amount types.Currency, useUnconfirmed bool, expiration time.Duration) ([]types.SiacoinElement, types.ChainIndex, types.Currency, error) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	knownAddresses := make(map[types.Address]bool)
+	relevantAddr := func(addr types.Address) bool {
+		if exists, ok := knownAddresses[addr]; ok {
+			return exists
+		}
+		_, err := m.store.WalletAddress(walletID, addr)
+		if errors.Is(err, ErrNotFound) {
+			knownAddresses[addr] = false
+			return false
+		} else if err != nil {
+			panic(err)
```

Comment:
Isn't a panic here a bit rough? Might be more forgiving to just return `false` and log the error.